pipeline/beam_tables.py

```diff
@@ -849,40 +851,22 @@ class ScanDataBeamPipelineRunner():
   """A runner to collect cloud values and run a corrosponding beam pipeline."""
 
   def __init__(self, project: str, bucket: str, staging_location: str,
-               temp_location: str, caida_ip_metadata_class: type,
-               caida_ip_metadata_bucket_folder: str, maxmind_class: type,
-               maxmind_bucket_folder: str, dbip_class: type,
-               dbip_bucket_folder: str) -> None:
+               temp_location: str,
+               metadata_prod_values: IpMetadataValuesInterface) -> None:
     """Initialize a pipeline runner.
 
     Args:
       project: google cluod project name
       bucket: gcs bucket name
       staging_location: gcs bucket name, used for staging beam data
       temp_location: gcs bucket name, used for temp beam data
-      caida_ip_metadata_class: an IpMetadataInterface subclass (class, not instance)
-      caida_ip_metadata_bucket_folder: gcs folder with CAIDA ip metadata files
-      maxmind_class: an IpMetadataInterface subclass (class, not instance)
-      maxmind_bucket_folder: gcs folder with maxmind files
-      dbip_class: a DbipMetadata class (class, not instance)
-      dbip_bucket_folder: gcs folder with dbip files
+      metadata_prod_values: values used to initialize an IpMetadataChooser
```

Comment:
There's no need for a `IpMetadataValues` and a `IpMetadataChooser`.
Just pass a `IpMetadataChooser`.
I imagine that the problem is that it has to get initialized in each worker. You can do lazy initialization on the first lookup.
Alternatively, pass a factory function that will return an initialized `IpMetadataChooser` given a date. The `IpMetadataValues` is an implementation detail that this code shouldn't need to know.

Reply:
made IpMetadataChooser contain the values and have a lazy init option.